reference/checkouts.v3.yml

```diff
@@ -8863,6 +8960,10 @@ components:
           type: string
           description: Key name to identify the type of coupon.
           example: percentage_discount
+        display_name:
+          type: string
+          description: The display name of the coupon.
+          example: 20% Off
```

Comment:
`coupons`

in `checkout.v3.yml` file, the main change is the `coupons` and `cart.coupons`, other changes are all about examples (we write exampled one by one, not use a reference to get rid of duplications, so quite a lot of examples to change)